source/magpie/dataframe.d

```diff
@@ -1098,6 +1098,68 @@ public:
                 data[i][pos] = elements.data[i];
         }
     }
+
+    /++
+    void apply(alias Fn, int axis, T)(T index)
+    Description: Applies a function on a particular row or column
+    @params: Fn - Function
+    @params: axis - 0 for rows, 1 for columns
+    @params: index - integer or string indexes of rows
+    +/
+    void apply(alias Fn, int axis, T)(T index)
+        if(is(T == int[]) || is(T == string[][]))
+    {
+        if(index.length == 0)
+            return;
+
+        static if(is(T == string[][]))
+            foreach(i; index)
+                assert(i.length == indx.indexing[axis].codes.length, "Index level mismatch");
+
+        int[] pos;
+        static if(is(T == int[]))
+            pos = index;
+        else
+        {
+            import std.array: appender;
+            auto loc = appender!(int[]);
```

Comment:
Appender is good for when you don't know how much you are going to append and you know there will be multiple concatenations. 

In this case you know ahead of time: `index.length`. Just do 
```d
pos.length = index.length;
foreach(i; index)
    pos[i] = getPosition!(axis)(i);
```
 and it will reserve the capacity for you.

Reply:
Addressed in dec86d1